src/control/IntersectionReferenceSystem.ts

```diff
@@ -92,7 +94,14 @@ export class IntersectionReferenceSystem {
     } else {
       this.endVector = IntersectionReferenceSystem.getDirectionVector(this.interpolators.trajectory, 1 - THRESHOLD_DIRECTION_DISTANCE, 1);
     }
-    this.startVector = this.endVector.map((d: number) => d * -1);
+
+    if (isFinite(trajectoryAngleStart)) {
+      const angleInRad = radians(trajectoryAngleStart);
+      const dirVector = new Vector2(Math.cos(angleInRad), Math.sin(angleInRad)).toArray();
+      this.startVector = dirVector;
+    } else {
+      this.startVector = IntersectionReferenceSystem.getDirectionVector(this.interpolators.trajectory, 0 + THRESHOLD_DIRECTION_DISTANCE, 0);
```

Comment:
Calculating direction vectors at both ends of a position log is quite risky if they are used for anything. Typically in real world several hundred meters of wellbore is very close to vertical but maybe not quite. So you risk getting a null vector or a pretty random direction. There is also a change in behavior here from startVector being the opposite of end vector to being either passed in or calculated, old behavior is not an option.

Reply:
Thanks for your feedback. I removed calculation of the start vector and continue using the opposite of end vector.

Reply:
old behavior is not back

Reply:
sorry. I was going to say "old behavior is **now** back".